Can't get it to work?

I have been trying to use this parser but even when I try the example code mentioned here, I can't get it to run. Any valid xml results in an invalid xml error because of EOF? What am I doing wrong?

>    
```
        f, _ := os.Open("input.xml")
	br := bufio.NewReader(f)
	parser := xmlparser.NewXMLParser(br, "book", "journal")

	for xml := range parser.Stream() {
		if xml.Err !=nil {
			fmt.Println("xml.Err:", xml.Err);
                        return;
		}

		fmt.Println(xml.Childs["title"][0].InnerText)
		if xml.Name == "book" {
			fmt.Println(xml.Childs["comments"][0].Childs["userComment"][0].Attrs["rating"])
			fmt.Println(xml.Childs["comments"][0].Childs["userComment"][0].InnerText)
		}
	}
```
>    
```
Output: 
xml.Err: EOF
```

Test cases:

> 

```
github.com/tamerh/xml-stream-parser$ go test
--- FAIL: TestBasics (0.00s)
panic: Test failed result must be 2 [recovered]
	panic: Test failed result must be 2

goroutine 5 [running]:
testing.tRunner.func1(0xc0000ae100)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x532bc0, 0x58a0c0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/tamerh/xml-stream-parser.TestBasics(0xc0000ae100)
	/go/src/github.com/tamerh/xml-stream-parser/xmlparser_test.go:38 +0x326d
testing.tRunner(0xc0000ae100, 0x5728a8)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x35c
exit status 2
FAIL	github.com/tamerh/xml-stream-parser	0.004s
```

go version go1.11.5 linux/amd64

--- a/xmlparser.go
+++ b/xmlparser.go
@@ -524,7 +524,6 @@ func (x *XMLParser) isCDATA() (bool, []byte, error) {
 	}
 
 	if b[0] != '!' {
-		x.unreadByte()
 		return false, nil, nil
 	}
 

